### PR TITLE
Global flow and task retries and retry delay seconds

### PR DESCRIPTION
**Pull request by robfreedy**
This PR exposes 4 new global settings that can be set: 

PREFECT_FLOW_DEFAULT_RETRIES,
PREFECT_FLOW_DEFAULT_RETRY_DELAY_SECONDS,
PREFECT_TASK_DEFAULT_RETRIES,
PREFECT_TASK_DEFAULT_RETRY_DELAY_SECONDS

These values do not overwrite retry or retry delay seconds that are set on the flow/task level. These values will be used if the flow/task does not have a retry/retry delay seconds value set. The default for all of these is 0.

This would solve: https://github.com/PrefectHQ/prefect/issues/9520.
### Example

```
prefect config set PREFECT_FLOW_DEFAULT_RETRIES=2
prefect config set PREFECT_TASK_DEFAULT_RETRIES=2
prefect config set PREFECT_FLOW_DEFAULT_RETRY_DELAY_SECONDS = [1, 10, 100]
prefect config set PREFECT_TASK_DEFAULT_RETRY_DELAY_SECONDS = [1, 10, 100]
```
If retries/retry delay seconds are not set, these values would be used
### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`, `docs`.

### CODE REVIEW

**Comment by billpalombi** [comment]
FWIW I support adding this functionality. Thanks @robfreedy.

**Comment by madkinsz** [comment]
@robfreedy can you fill out the body of the pull request and update the docs for this?

File: src/prefect/tasks.py [hunk 1]
```diff
@@ -242,7 +248,12 @@ def __init__(
         # TaskRunPolicy settings
         # TODO: We can instantiate a `TaskRunPolicy` and add Pydantic bound checks to
         #       validate that the user passes positive numbers here
-        self.retries = retries
+        print(retries)
```
Comment (madkinsz) [inline src/prefect/tasks.py:251]: Needs to be removed

File: tests/test_flows.py [hunk 2]
```diff
@@ -2035,6 +2036,21 @@ def parent_flow():
             child_flow_run_count == 4
         ), "Child flow should run 2 times for each parent run"
 
+    def test_global_retry_config(self):
```
Comment (madkinsz) [inline tests/test_flows.py:2039]: Missing test coverage for `PREFECT_FLOW_DEFAULT_RETRY_DELAY_SECONDS`

File: tests/test_tasks.py [hunk 3]
```diff
@@ -976,6 +977,41 @@ def test_flow():
                     last_context.start_time < context.start_time
                 ), "Timestamps should be increasing"
 
+    @pytest.mark.parametrize("always_fail", [True, False])
```
Comment (madkinsz) [inline tests/test_tasks.py:980]: Missing test coverage for `PREFECT_TASK_DEFAULT_RETRY_DELAY_SECONDS`

File: tests/test_tasks.py [hunk 4]
```diff
@@ -976,6 +977,41 @@ def test_flow():
...
+            if always_fail:
+                assert task_run_state.is_failed()
+                assert exceptions_equal(
+                    await task_run_state.result(raise_on_failure=False), exc
+                )
+                assert mock.call_count == 4
+            else:
+                assert task_run_state.is_completed()
+                assert await task_run_state.result() is True
+                assert mock.call_count == 4
```
Comment (madkinsz) [inline tests/test_tasks.py:1013]: No need to test `always_fail` behavior here — that's a different test. All we care about here is the retry count.

File: src/prefect/settings.py [hunk 5]
```diff
@@ -641,6 +641,32 @@ def default_cloud_ui_url(settings, value):
...
+"""
+This value sets the retry delay seconds for all flows when this environment is set
+This value does not overwrite invidually set retry delay seconds
+"""
+
+PREFECT_TASK_DEFAULT_RETRY_DELAY_SECONDS = Setting(
+    Union[float, int, List[float], Callable[[int], List[float]]], default=0
+)
+"""
+This value sets the retry delay seconds for all flows when this environment is set
```
Comment (madkinsz) [inline src/prefect/settings.py:666]:
```suggestion
This value sets the retry delay seconds for all tasks when this environment is set
```

File: src/prefect/settings.py [hunk 6]
```diff
@@ -641,6 +641,32 @@ def default_cloud_ui_url(settings, value):
 task will refresh the cached results. Defaults to `False`.
 """
 
+PREFECT_TASK_DEFAULT_RETRIES = Setting(int, default=0)
+"""
+This value sets the retries for all tasks associated with a flow.
```
Comment (madkinsz) [inline src/prefect/settings.py:646]:
```suggestion
This value sets the default number of retries for all tasks.
```

File: src/prefect/settings.py [hunk 7]
```diff
@@ -641,6 +641,32 @@ def default_cloud_ui_url(settings, value):
...
 
+PREFECT_TASK_DEFAULT_RETRIES = Setting(int, default=0)
+"""
+This value sets the retries for all tasks associated with a flow. 
+This value does not overwrite individually set retries values on tasks
+"""
+
+PREFECT_FLOW_DEFAULT_RETRIES = Setting(int, default=0)
+"""
+This value sets the retries for all flows.
```
Comment (madkinsz) [inline src/prefect/settings.py:652]:
```suggestion
This value sets the default number of retries for all flows. 
```

File: src/prefect/settings.py [hunk 8]
```diff
@@ -641,6 +641,32 @@ def default_cloud_ui_url(settings, value):
...
+
+PREFECT_FLOW_DEFAULT_RETRIES = Setting(int, default=0)
+"""
+This value sets the retries for all flows. 
+This value does not overwrite individually set retries values on a flow
+"""
+
+PREFECT_FLOW_DEFAULT_RETRY_DELAY_SECONDS = Setting(Union[int, float], default=0)
+"""
+This value sets the retry delay seconds for all flows when this environment is set
```
Comment (madkinsz) [inline src/prefect/settings.py:658]:
```suggestion
This value sets the retry delay seconds for all flows.
```

File: src/prefect/settings.py (hunk 5 above)
Comment (madkinsz) [inline src/prefect/settings.py:666]:
```suggestion
This value sets the default retry delay seconds for all tasks.
```

File: src/prefect/settings.py (hunk 8 above)
Comment (madkinsz) [reply src/prefect/settings.py:658]: Really could just be "The default retry delay seconds for all flows." — "This value sets" is pretty redundant.

**Review by madkinsz — APPROVED** [review]

**Pull request merged**